### ISSUE TITLE
External Issue: LIEF fails to parse example MachO

### ISSUE THREAD

**Issue by jhhcs** [binref/refinery#59]
This issue tracks https://github.com/lief-project/LIEF/issues/1110, which is currently the main blocker for switching completely to LIEF as our executable parsing abstraction layer. This would replace 3 dependencies, some of them rather stale, with a single one that is well-maintained.

**Comment by ForensicITGuy** [comment]
This should now be fixed in LIEF 0.16 https://lief.re/doc/stable/changelog.html#december-10th-2024

**Comment by jhhcs** [comment]
Indeed, and that is good - now I have another issue. I am also depending on [SMDA](https://github.com/danielplohmann/smda), which [pins me to LIEF-0.14.0](https://github.com/danielplohmann/smda/blob/99559416efb5afd389f19b638d073a7457c98514/setup.py#L13).

**Comment by jhhcs** [comment]
Nice, this can be closed, LIEF branch looking really good now.
